homeassistant/components/nissan_leaf/binary_sensor.py

```diff
@@ -0,0 +1,44 @@
+"""Plugged In Status Support for the Nissan Leaf."""
+import logging
+
+from homeassistant.components.nissan_leaf import (
+    DATA_LEAF, DATA_PLUGGED_IN, LeafEntity)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['nissan_leaf']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up of a Nissan Leaf binary sensor."""
+    _LOGGER.debug(
+        "binary_sensor setup_platform, discovery_info=%s", discovery_info)
+
+    devices = []
+    for key, value in hass.data[DATA_LEAF].items():
+        _LOGGER.debug(
+            "binary_sensor setup_platform, key=%s, value=%s", key, value)
+        devices.append(LeafPluggedInSensor(value))
+
+    add_devices(devices, True)
+
+
+class LeafPluggedInSensor(LeafEntity):
+    """Plugged In Sensor class."""
+
+    @property
+    def name(self):
+        """Sensor name."""
+        return "{} {}".format(self.car.leaf.nickname, "Plug Status")
+
+    @property
+    def state(self):
```

Comment:
We should not overwrite the `state` property for this class. Use `is_on` instead.

Reply:
Fixed with commit https://github.com/filcole/home-assistant/commit/86dadb4147846afc6411cc9b44bd221ef35cb780